File: LearnJava/tydic-bi-meta/WebRoot/meta/module/reportManage/page.js
/******************************************************
 *Copyrights @ 2011，Tianyuan DIC Information Co., Ltd.
 *All rights reserved.
 *
 *Filename：
 *       favCommon.js
 *Description：
 *       分页功能js
 *Dependent：
 *       dhtmlx.js，dwr 有关JS，公共JS
 *Author:
 *       李国民
 *Date：
 *       12-3-20
 ********************************************************/

	var totalCount=0;		//查询总数
	var page=1;				//当前显示第几页（默认1）
	var pageCount=1;		//总共几页
	var count=7;			//每页展示几条（默认7）
	var pageBean ={};		//分页对象
		pageBean.posStart = 0;	
		pageBean.count = count;
	
	//设置分页值
	setPage = function(pageValue){
		page = pageValue;
		pageBean.posStart = (page-1)*count;
		pageBean.count = count;
	}
	
	//展现分页
	showPage = function(){
		var tem = totalCount/count+"";
		if(tem.indexOf(".")!=-1){
			pageCount = parseInt(tem*1)+1;
		}else{
			pageCount = tem*1;
		}
		var pageStr = "";
		if(pageCount==1){		//如果总分页数等于1表示没有分页
			pageStr += "首页&nbsp;&nbsp;上一页&nbsp;&nbsp;下一页&nbsp;&nbsp;末页&nbsp;&nbsp;&nbsp;&nbsp;";
			
		}else{
			if(page==1){  //第一页时，首页和上一页没有数据
				pageStr += "首页&nbsp;&nbsp;上一页&nbsp;&nbsp;";
			}else{
				pageStr += "<a onClick='doPage(1)' style='color:blue;'>首页</a>&nbsp;&nbsp;";
				pageStr += "<a onClick='doPage("+(page-1)+")' style='color:blue;'>上一页</a>&nbsp;&nbsp;";
			}
			if(page==pageCount){
				pageStr += "下一页&nbsp;&nbsp;末页&nbsp;&nbsp;&nbsp;&nbsp;";
			}else{
				pageStr += "<a onClick='doPage("+(page+1)+")' style='color:blue;'>下一页</a>&nbsp;&nbsp;";
				pageStr += "<a onClick='doPage("+pageCount+")' style='color:blue;'>末页</a>&nbsp;&nbsp;&nbsp;&nbsp;";
			}
		}
		pageStr += "每页显示"+count+"条&nbsp;&nbsp;总共"+totalCount+"条&nbsp;&nbsp;当前"+page+"/"+pageCount+"页";
		return pageStr;
	}

File: LearnJava/tydic-bi-meta/src/tydic/meta/common/Page.java
package tydic.meta.common;

import java.util.HashMap;
import java.util.Map;

/**
 * Copyrights @ 2011,Tianyuan DIC Information Co.,Ltd. All rights reserved.<br>
 * @author 张伟
 * @description 作用:分页PO对象
 * @date 2011-9-15
 */
public class Page {

	private static Page instance = null;
	
    /**
     * 起始行数
     */
    private int posStart=0;
    /**
     * 每页记录数
     */
    private int count=15;

    public Page() {
    
    }
   /**
    * 单例模式实例化
    */
   public synchronized static Page getInstance() {
		if (instance == null) {
			instance = new Page();
		}
		return instance;
	}
   
    public Page(int posStart, int count) {
        this.posStart = posStart;
        this.count = count;
    }

    public int getPosStart() {
        return posStart;
    }

    public void setPosStart(int posStart) {
        this.posStart = posStart;
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }
    /**
     * @author qx
     * @description 分页
     * @param params
     */
    public Map getPageInfo(Map params){
     Map page = new HashMap() ;
     int start,end;
     //数据总数
     int totalSize =Integer.parseInt(params.get("totalSize").toString());
     //当前页面 从0开始
     int pageIndex =Integer.parseInt(params.get("pageIndex").toString());
     //页面展现数据数量
     int pageSize =Integer.parseInt(params.get("pageSize").toString());
     pageSize = (pageSize < 1) ? 20 : pageSize;
     
     int pageCount = totalSize/pageSize+1;
     if (totalSize <= 0) {
      pageCount = 0;
      start = 0;
      end = pageSize;
      pageIndex = 0;
      totalSize = 0;
     }else{
      pageCount = totalSize / pageSize;
      pageCount = (totalSize % pageSize == 0) ? pageCount : pageCount + 1;
      pageIndex = (pageIndex < 0) ? 0 : pageIndex;
      pageIndex = (pageIndex >= pageCount) ? pageIndex - 1 : pageIndex;
      // 计算开始、结束记录行号
      start = pageIndex * pageSize + 1;
      end = (pageIndex + 1) * pageSize;
     }
     page.put("pageCount", pageCount);
     page.put("start", start);
     page.put("end", end);
     page.put("pageNo", pageIndex);
     page.put("total", totalSize);
     page.put("start", start);
     page.put("end", end);
     return page;
    }
}
